src/ServiceControl.Monitoring/QueueLength/DefaultQueueLengthProvider.cs

```diff
@@ -0,0 +1,40 @@
+﻿namespace ServiceControl.Monitoring.QueueLength
+{
+    using System.Threading.Tasks;
+    using Infrastructure;
+    using Messaging;
+    using NServiceBus.Metrics;
+
+    public class DefaultQueueLengthProvider : IProvideQueueLength
```

Comment:
This class looks as if it was prepared for being a base class for any transport specific implementation. 
I think this might be a bit premature.

Reply:
`virtual` + `protected` without inheritors? I agree with @tmasternak

Reply:
Fair enough. It was just transferred from the other PR. It _is_ intended to be a base class for any transport specific implementation. When I wrote it, it was an abstract base class and I created an MSMQ specific implementation over the top. Then I realized that the MSMQ implementation would have no code in it and that we don't need an MSMQ specific version.